tests/test_keys/test_umbral_keys.py

```diff
@@ -9,6 +9,32 @@ def test_gen_key():
     umbral_pub_key = umbral_priv_key.get_pubkey()
     assert type(umbral_pub_key) == keys.UmbralPublicKey
 
+def test_derive_key_from_label():
+    master_secret = b"random master secret"
```

Comment:
This is a fine test in that it shows a key can be derived from arbitrary data a la `master_secret`.

However, I don't think this test is showing proper or secure behavior. A user might think that they can derive a key from low entropy data such as an alphanumeric password.

I don't think this is the secure/proper way to do this, see the actual derivation function below for an alternative.

Reply:
Agree.

Reply:
What if we do something like:
```
b'very secure master seed approve recall age route small fine gospel depend canvas inherit impulse people'
```
(yes, this was a secure seed until I posted it)